server/services/user.service.js

```diff
@@ -245,18 +246,31 @@ export const updateUser = async (updatedUser) => {
   }
 };
 
-export const getAllRegisteredUsers = async () =>
-  User.aggregate([
+export const getAllRegisteredUsers = async (pageNumber, resultLimit) => {
```

Comment:
I think we can use the page and limit as default parameters directly and remove the two lines below
```suggestion
export const getAllRegisteredUsers = async (page = 1, limit=10) => {
```